sdk/framework/openapi.go

```diff
@@ -464,6 +465,21 @@ func specialPathMatch(path string, specialPaths []string) bool {
 func expandPattern(pattern string) []string {
 	var paths []string
 
+	// recursively operate on each pattern after determining a set of unique roots is present,
+	// e.g. "(rootOne|rootTwo)/(?P<name>regex)" should yield "rootOne/{name}", "rootTwo/{name}"
```

Comment:
It may be a good idea to extract this logic into a separate function and add a simple unit test to verify it